pkg/logql/log/parser_test.go

```diff
@@ -111,6 +111,73 @@ func Test_jsonParser_Parse(t *testing.T) {
 	}
 }
 
+func TestKeyShortCircuit(t *testing.T) {
+	simpleJsn := []byte(`{
+      "data": "Click Here",
+      "size": 36,
+      "style": "bold",
+      "name": "text1",
+      "name": "duplicate",
+      "hOffset": 250,
+      "vOffset": 100,
+      "alignment": "center",
+      "onMouseUp": "sun1.opacity = (sun1.opacity / 100) * 90;"
+    }`)
+	logFmt := []byte(`data="ClickHere" size=36 style=bold name=text1 name=duplicate hOffset=250 vOffset=100 alignment=center onMouseUp="sun1.opacity = (sun1.opacity / 100) * 90;"`)
+
+	hints := &fakeParseHints{label: "name"}
```

Comment:
Why not used the actually ParserHints struct here?

Reply:
I try to use mocks in unit tests where possible so the test isn't couple to any of the behavior of the actual parser hints. As long as the interface behaves as it should, this test doesn't ever need to change because the way we implement ParserHints changes. It also means that we don't have to set this test up with anything specific to parser hints.